backend/inspirehep/hal/tasks.py

```diff
@@ -0,0 +1,96 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2019 CERN.
+#
+# inspirehep is free software; you can redistribute it and/or modify it under
+# the terms of the MIT License; see LICENSE file for more details.
+
+
+import structlog
+from celery import shared_task
+from celery.exceptions import SoftTimeLimitExceeded
+from requests.exceptions import RequestException
+from time_execution import time_execution
+
+from inspirehep.hal.core.sword import create, update
+from inspirehep.hal.core.tei import convert_to_tei
+
+LOGGER = structlog.getLogger()
+
+
+# `soft_time_limit` is used to schedule a retry.
+# `time_limit` instead kills the worker process (its exception cannot be caught
+# in order to schedule a retry).
+# Race conditions between the 2 limits are possible, but the orcid push
+# operation is idempotent.
+@shared_task(bind=True, soft_time_limit=10 * 60, time_limit=11 * 60)
+@time_execution
+def hal_push_task(self, record):
+    """Celery task to push a record to HAL.
+
+    Args:
+        self (celery.Task): the task
+        rec_id (Int): inspire record to push to HAL.
+    """
+    LOGGER.info("New hal_push task", recid=record["control_number"])
+    try:
+        _hal_push(record)
+        LOGGER.info(
+            "hal_push task successfully completed.", recid=record["control_number"]
+        )
+    except (RequestException, SoftTimeLimitExceeded) as exc:
+        # Enrich exception message.
+        if isinstance(exc, RequestException):
+            message = (exc.args[0:1] or ("",))[0]
+            message += "\nResponse={}".format(
+                exc.response.content if hasattr(exc, "response") else None
+            )
+            message += "\nRequest={} {}".format(exc.request.method, exc.request.url)
+            exc.args = (message,) + exc.args[1:]
+
+        # If max_retries=3, then self.request.retries is: [0, 1, 2, 3]
+        # thus backoff power 4 is (secs): [4*60, 16*60, 64*60]
+        backoff = (4 ** (self.request.retries + 1)) * 60
+
+        LOGGER.warning(
+            f"hal_push task raised an exception. Retrying in {backoff} secs.",
+            recid=record["control_number"],
+        )
+        raise self.retry(max_retries=3, countdown=backoff, exc=exc)
+    except Exception:
+        LOGGER.warning("hal_push task failed", recid=record["control_number"])
+        raise
```

Comment:
actually I don't think you need any of this, and can simply use celery auto-retry on some specific exceptions, like we're doing in https://github.com/inspirehep/inspirehep/blob/9878076d6f7389fbb4358ab553f0a7e620fd31b5/backend/inspirehep/indexer/tasks.py#L58-L60

I think the ORCID stuff was written before we were using a compatible celery version and we never changed it.